[RHDX-160] Taxonomy Term List

rogeralloymagnetic opened the pull request:
### JIRA Issue Link
* https://projects.engineering.redhat.com/browse/RHDX-160

### Verification Process
**Note:** All steps below have been completed and can be quickly viewed here: 
https://developer-preview-3538.ext.us-west.dc.preprod.paas.redhat.com/taxonomy/term/39

1. Choose a taxonomy term to use for testing. I used the Control Tag "Containers" (taxonomy term id: 39)
2. Go to one item of content for all of the Content Types listed below and add the taxonomy term you chose to that piece of content, then save. At this point you will probably want to add or remove parts of the content you want to test (ie: with or without an image) as well as change the Published date to today so that it shows up at the top of the list for easier viewing.
3. Go to the taxonomy term page for the term you chose. ie: /taxonomy/term/39
4. Ensure the page matches the mock on the ticket.

### Updated Content Types
- Article
- eBook
- Cheat Sheet
- Coding Resource
- Events
- Katacoda Course
- Katacoda Individual Lesson
- Page (assembly_page)
- Product
- Video Resource

### Screenshots
#### Desktop (Part 1)
![Screen Shot 2020-04-02 at 1 19 34 PM](https://user-images.githubusercontent.com/41014351/78295709-ce908080-74e9-11ea-93fc-501ebb8a7160.png)

#### Desktop (Part 2)
![Screen Shot 2020-04-02 at 1 19 48 PM](https://user-images.githubusercontent.com/41014351/78295708-cd5f5380-74e9-11ea-9407-718d580c9c36.png)

#### Mobile
![Screen Shot 2020-04-02 at 1 21 02 PM](https://user-images.githubusercontent.com/41014351/78295701-cafcf980-74e9-11ea-93ad-c2b382642cef.png)

Commit:
Start taxonomy list content modeling.

Commit:
Merge branch 'master' into RHDX-160

Commit:
Start first taxonomy list item template.

Commit:
taxonomy-list-item for articles, books, cheat sheets and katacodas.

Commit:
taxonomy-list-item for video_resources.

Commit:
taxonomy-list-view for products.

Commit:
Refactor author links.

Commit:
taxonomy-list-item mobile style fixes, image updates, and remove unused code.

Commit:
taxonomy_term view pager and RSS.

Commit:
Merge branch 'master' into RHDX-160

alistairmc commented:
rebuild this please

rogeralloymagnetic commented:
rebuild this please

rogeralloymagnetic commented:
Rebuild this please

rogeralloymagnetic commented:
I'll run the e2e tests locally to see if I can reproduce the failure. In the meantime...

rogeralloymagnetic commented:
retest this please

jordanpagewhite commented:
retest this please

Commit:
Export conflicting config.

jordanpagewhite commented:
@rogeralloymagnetic Please check config: https://developer-preview-3538.ext.us-west.dc.preprod.paas.redhat.com/admin/config/development/configuration

rogeralloymagnetic commented:
rebuild this please

rogeralloymagnetic commented:
rebuild this please

jordanpagewhite commented:
rebuild this please

rogeralloymagnetic commented:
rebuild this please

rogeralloymagnetic commented:
rebuild this please

jordanpagewhite commented:
rebuild this please

staceymosier commented:
rebuild this please

jordanpagewhite commented:
rebuild this please

jordanpagewhite commented:
rebuild this please

jordanpagewhite commented:
rebuild this please

Commit:
Merge branch 'master' into RHDX-160

staceymosier reviewed (changes requested):
Can you please check the positioning of the `rhd-c-taxonomy-list__tag`?

<img width="1112" alt="Screen Shot 2020-04-14 at 10 46 07 AM" src="https://user-images.githubusercontent.com/19995927/79256628-31b0da00-7e3d-11ea-9905-8b6b9863e48e.png">

Commit:
Fix tag positioning in taxonomy term lists.

gdoyle1 reviewed (approved):
Works as expected with the exception of the bugs you documented in the comments.
Icons for content types we discussed:
1. Product
fas fa-box-open (https://fontawesome.com/icons/box-open?style=solid)
2. Page
fas fa-file-alt (https://fontawesome.com/icons/file-alt?style=solid)
3. Event
fas fa-calendar (https://fontawesome.com/icons/calendar?style=solid)

sjcox-rh reviewed (approved):
@rogeralloymagnetic all looks good outside of the bugs you documented. Let me know if you need any additional default images.  Thanks!

Commit:
Add all Purpose Attributes to the taxonomy_term View. Add taxonomy-lit-item templates for Events and Pages.

Commit:
Update default images for Taxonomy Term Items

Commit:
Add coding_resource content type to taxonomy_list_item.

Commit:
Add default image to assembly_page content type for taxonomy_list_item.

Commit:
Add title url and presenters to events in taxonomy_list_item.

Commit:
events taxonomy_term_list config.

sjcox-rh commented:
@rogeralloymagnetic just did a final UX text on all the content types and everything looks good to go!

Commit:
Merge branch 'master' into RHDX-160

Commit:
Add access check to author nodes in _generate_authors_links().

rogeralloymagnetic commented:
rebuild this please

alistairmc commented:
rebuild this please

rogeralloymagnetic commented:
rebuild this please

staceymosier commented on _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme:
```diff
@@ -396,6 +435,11 @@ function rhdp2_preprocess_node(&$variables) {
       $variables['timestamp'] = $timestamp;
     }
+    
+    // Prepares the article authors render element for taxonomy page.
```
There are 3 calls to _generate_authors_links split up across the file here, I have to jump to each 'if content type'
 check to read this.

I would rather see this collected into one code block,  for example line 151 for similar pattern
`if (in_array($type, $measured_read_time_types) && $variables['view_mode'] == 'learning_path_card')`
so that I don't have to jump to each content-type check to read the same code.

Also, please check PHPCS, "Found 7 errors and 1 warning" for rhdp2.theme.

rogeralloymagnetic replied:
@staceymosier Resolved in 3c243f5

Commit:
Refactor taxonomy_list_item view_mode. Fix phpcs errors in rhdp2.theme.

Commit:
Merge branch 'master' into RHDX-160

rogeralloymagnetic commented:
Retest this please.

jordanpagewhite commented:
rebuild this please

jordanpagewhite commented:
rebuild this please

jordanpagewhite commented:
https://developer-preview-3538.ext.us-west.dc.preprod.paas.redhat.com/

rogeralloymagnetic commented:
Retest this please.

jordanpagewhite commented:
rebuild this please

jordanpagewhite commented:
retest this please

jordanpagewhite committed:
Add field_sections to hidden taxonomy_list_item

This adds field_sections to the hidden entity_view_display setting for
the katacoda_individual_lesson.taxonomy_list_item field. This field was
added in RHDX-457 (#3555) and merged into this branch via master.

jordanpagewhite commented:
rebuild this please

rogeralloymagnetic commented:
retest this please

rogeralloymagnetic commented:
Retest this please.

Commit:
Merge branch 'master' into RHDX-160

Commit:
Export config.

Commit:
Merge branch 'RHDX-160' of github.com:rogeralloymagnetic/developers.redhat.com into RHDX-160

jordanpagewhite commented:
I disabled the Drupal: Pull Request Build so that we could merge this PR. That status check will likely fail on all future attempts on all PRs because of the 1 hour timeout.
